src/native/managed/cdacreader/src/Contracts/RuntimeTypeSystem_1.cs

```diff
@@ -487,10 +540,194 @@ public MethodDescHandle GetMethodDescHandle(TargetPointer methodDescPointer)
         Data.MethodDescChunk validatedMethodDescChunkData = _target.ProcessedData.GetOrAdd<Data.MethodDescChunk>(methodDescChunkPointer);
         Data.MethodDesc validatedMethodDescData = _target.ProcessedData.GetOrAdd<Data.MethodDesc>(methodDescPointer);
 
-        MethodDesc trustedMethodDescF = new MethodDesc(methodDescPointer, validatedMethodDescData, validatedMethodDescChunkData);
+        MethodDesc trustedMethodDescF = new MethodDesc(_target, methodDescPointer, validatedMethodDescData, validatedMethodDescChunkData);
         _ = _methodDescs.TryAdd(methodDescPointer, trustedMethodDescF);
         return new MethodDescHandle(methodDescPointer);
     }
 
     public TargetPointer GetMethodTable(MethodDescHandle methodDescHandle) => _methodDescs[methodDescHandle.Address].MethodTable;
+
+    public bool IsGenericMethodDefinition(MethodDescHandle methodDescHandle)
+    {
+        MethodDesc methodDesc = _methodDescs[methodDescHandle.Address];
+
+        if (methodDesc.Classification != MethodDescClassification.mcInstantiated)
+            return false;
+
+        Data.InstantiatedMethodDesc instantiatedMethodDesc = _target.ProcessedData.GetOrAdd<Data.InstantiatedMethodDesc>(methodDescHandle.Address);
+        return ((int)instantiatedMethodDesc.Flags2 & (int)InstantiatedMethodDescFlags2.KindMask) == (int)InstantiatedMethodDescFlags2.GenericMethodDefinition;
+    }
```

Comment:
I think we can be a bit more high level here:

I really don't like this pattern for the MethodDesc subclasses.

I've been doing this in my fork:
```
    // member of RuntimeTypeSystem_1
    internal InstantiatedMethodDesc AsInstantiatedMethodDesc(MethodDesc md)
    {
        Debug.Assert(md.Classification == MethodClassification.Instantiated);
        Data.InstantiatedMethodDesc imd = _target.ProcessedData.GetOrAdd<Data.InstantiatedMethodDesc>(md.Address);
        return new InstantiatedMethodDesc(md.Address, imd);
    }
```

where `InstantiatedMethodDesc` is a wrapper around `Data.InstantiatedMethodDesc`:

```csharp
  // also in RuntimeTypeSystem_1
      internal struct InstantiatedMethodDesc
    {
        private readonly TargetPointer _address;
        private readonly Data.InstantiatedMethodDesc _desc;
        internal InstantiatedMethodDesc(TargetPointer address, Data.InstantiatedMethodDesc instantiatedMethodDesc)
        {
            _address = address;
            _desc = instantiatedMethodDesc;
        }

        private bool HasFlags(InstantiatedMethodDescFlags mask, InstantiatedMethodDescFlags flags) => (_desc.Flags2 & (ushort)mask) == (ushort)flags;
        internal bool IsWrapperStubWithInstantiations => HasFlags(InstantiatedMethodDescFlags.KindMask, InstantiatedMethodDescFlags.WrapperStubWithInstantiations);
        internal bool IsGenericMethodDefinition => HasFlags(InstantiatedMethodDescFlags.KindMask, InstantiatedMethodDescFlags.GenericMethodDefinition);

        internal bool HasPerInstInfo => _desc.PerInstInfo != TargetPointer.Null;

        internal bool HasMethodInstantiation => IsGenericMethodDefinition || HasPerInstInfo;

    }
```